board/app/models/comment.php

```diff
@@ -1,26 +1,195 @@
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $params = array(
+                $this->id,
+                $this->user_id
+            );
+            $db->query('DELETE FROM comment WHERE id = ? AND user_id = ?', $params);
+            $this->deleteFavoritedComment();
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    /*
+    * Functions with regards to favorite/unfavorite comments
+    */
+
+    public function countFavorite() 
+    {
+        $db = DB::conn();
+        $total_favorite = $db->value('SELECT COUNT(*) FROM favorite WHERE comment_id =?', array($this->id));
+    
+        return $total_favorite;
+    }
+
+    public function deleteFavoritedComment() 
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $params = array(
+                $this->id,
+                $this->user_id
+            );
+            $db->query('DELETE FROM favorite WHERE comment_id = ? AND user_id = ?', $params );
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    public function isUserComment()     
+    {
+        return $this->user_id === $_SESSION['user_id'];       
+    }
+
+    public function addFavorite()
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $params = array(
+                'comment_id' => $this->id,
+                'comment_body'=> $this->body, 
+                'user_id' => $this->user_id
+            );
+            $db->insert('favorite', $params);
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    public function removeFavorite() 
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $params = array(
+                /*if you removed this, once you unvaforited a comment, 
+                every comment you favorited will be deleted*/
+                $this->id, 
+                $this->user_id
+            );
+            $db->query('DELETE FROM favorite WHERE comment_id = ? AND user_id = ?', $params);
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    public function isCommentFavorited() 
+    {
+        $db = DB::conn();
+        $params = array(
+            $this->id,
+            $_SESSION['user_id']
+        );
+        $comment_favorited = $db->rows('SELECT * FROM favorite WHERE comment_id = ? AND user_id = ?', $params);
+        return !$comment_favorited;
+    }
+
+    public static function getAllFavorites($offset, $limit, $user_id) 
+    {
+        $favorites = array();
+        $db = DB::conn();
+                        
+        $rows = $db->rows("SELECT * FROM favorite WHERE user_id = ? LIMIT {$offset}, {$limit}", array($user_id));
+
+            foreach($rows as $row) {
```

Comment:
:scream: 
Hehe, :laughing:  done po. Thank you :bow:
-->https://github.com/maragatchalian/FoodFacts/commit/96d26e82c9eff60c909847a48eb896ba79346448